src/app/python/core.py

```diff
@@ -330,5 +413,30 @@ def _notify_event(self, f):
 
     @param f: The notification callable
     """
-    with self._core_listeners_lock:
-      self._core_listeners_strategy.send(f, self._core_listeners)
+    self._core_listeners_strategy.send(f, self._core_listeners)
+
+
+class CoreThread(Thread):
+  """
+  Core app Thread
+  """
+
+  def __init__(self, app):
```

Comment:
@grondinjc We'll nedd more stuff from the app later (like the apply_changes method)